content/BCBestPractices/Performance/DeleteAll/index.md

```diff
@@ -0,0 +1,32 @@
++++
+title = "DeleteAll"
+weight = 1180
++++
+
+<_Created by waldo, Described by waldo_\>
+
+## Description
+
+When you perform a "DeleteAll" when there is nothing to delete, it will still perform a lock.  When you for example perform a DeleteAll on an empty table, it will result in a table lock.
+Therefore it's good practice to always check if the table is empty when performing a DeleteAll.
+
+## Bad code
+
+```al
+    EmptyTableWLD.SetRange(Code, 'AJ');
+    EmptyTableWLD.DeleteAll(true);
+```
+
+## Good code
+
+```al
+    EmptyTableWLD.SetRange(Code, 'AJ');
+    if not EmptyTableWLD.IsEmpty() then
+        EmptyTableWLD.DeleteAll(true);
+```
+
+## [Discussions](https://github.com/microsoft/alguidelines/discussions/categories/bc-best-practices?discussions_q=DeleteAll+category%3A%22BC+Best+Practices%22)
```

Comment:
Do You really want the discussion link to be a header?

Reply:
Yeah - that doesn't really make sense, I agree .. :(